PowerEditor/src/WinControls/Preference/preference.rc

```diff
@@ -323,11 +323,12 @@ STYLE DS_SETFONT | DS_FIXEDSYS | DS_CONTROL | WS_CHILD
 FONT 8, "MS Shell Dlg", 0, 0, 0x1
 BEGIN
     CONTROL         "Don't fill find field in Find dialog with selected word", IDC_CHECK_STOPFILLINGFINDFIELD, "Button", BS_AUTOCHECKBOX | WS_TABSTOP,37,10,350,10
-    CONTROL         "Use Monospaced font in Find dialog (Need to restart Notepad++)",IDC_CHECK_MONOSPACEDFONT_FINDDLG, "Button", BS_AUTOCHECKBOX | WS_TABSTOP,37,25,350,10
-    CONTROL         "Find dialog remains open after search that outputs to results window",IDC_CHECK_FINDDLG_ALWAYS_VISIBLE, "Button", BS_AUTOCHECKBOX | WS_TABSTOP,37,40,350,10
-    CONTROL         "Confirm Replace All in All Opened Documents",IDC_CHECK_CONFIRMREPLOPENDOCS, "Button", BS_AUTOCHECKBOX | WS_TABSTOP,37,55,350,10
-    CONTROL         "Replace: Don't move to the following occurrence", IDC_CHECK_REPLACEANDSTOP, "Button", BS_AUTOCHECKBOX | WS_TABSTOP, 37, 70, 350, 10
-    CONTROL         "Search Result window: show only one entry per found line", IDC_CHECK_SHOWONCEPERFOUNDLINE, "Button", BS_AUTOCHECKBOX | WS_TABSTOP, 37, 85, 350, 10
+    CONTROL         "Don't select closest word when Find dialog is displayed", IDC_CHECK_STOPSELECTINGNEARESTWORDONFIND, "Button", BS_AUTOCHECKBOX | WS_TABSTOP, 37, 25, 350, 10

```

Comment:
Suggest reversing logic (all these "don't"s make my head hurt when trying to think about what it might do), and change wording to "Auto-fill Find-What with word at caret when no selection".

Of course, such a suggestion practically begs for the existing _Don't fill find field in Find dialog with selected word_ to be changed as well.

I'd suggest:

(checkbox) **Auto-fill Find-What when searching is invoked**
and then directly underneath with a slight right indentation:
(checkbox) **Use word at caret when no selection**

Both checkboxes would be ticked by default, preserving how N++ has operated for years.

Call me a dreamer.  :-)

Reply:
Thank you for the good suggestions @alankilborn .  I wasn't 100% happy with the wording either.  The only problem with reversing the logic of the current checkbox is how to deal with saved preferences from older versions of notepad++?  I could look for the old preference name and use the reversed version of the value if it is there but that is kind of messy.  What do you think?

Reply:
Well, if my suggestion is REALLY liked (and even liked by @donho, a dubious prospect :-) ), then I'd say don't try to reuse what is already there, and let the existing setting "die off" and create two new ones.  This would have the advantage of clarity going forward, but the disadvantage of forcing users that have already made a non-default choice for the existing setting to come back and make a new choice -- but this could actually be an advantage because then they would be aware of the improved possible choices!

Reply:
@alankilborn 
> Call me a dreamer. :-)

What do you think the following GUI:
![image](https://user-images.githubusercontent.com/90293/186736627-721492f4-0edc-44b6-9e0b-09b16a7be729.png)

Do you guys think it needs to be reworded in order to make it more clear?

(RC code)
```
IDD_PREFERENCE_SUB_SEARCHING DIALOGEX 0, 0, 455, 185
STYLE DS_SETFONT | DS_FIXEDSYS | DS_CONTROL | WS_CHILD
FONT 8, "MS Shell Dlg", 0, 0, 0x1
BEGIN
    GROUPBOX        "When launch Find dialog",IDD_FILLFINDFIEL_GRPSTATIC,31,4,323,43,BS_CENTER
    CONTROL         "Select the word on which caret is",IDC_CHECK_FILLFINDFIEL_WITHSELECTED,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,52,31,275,10
    CONTROL         "Fill find field with selected word",IDC_CHECK_FILLFINDFIEL_SELECT_CARET,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,16,275,10
    CONTROL         "Use Monospaced font in Find dialog (Need to restart Notepad++)",IDC_CHECK_MONOSPACEDFONT_FINDDLG,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,52,350,10
    CONTROL         "Find dialog remains open after search that outputs to results window",IDC_CHECK_FINDDLG_ALWAYS_VISIBLE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,67,350,10
    CONTROL         "Confirm Replace All in All Opened Documents",IDC_CHECK_CONFIRMREPLOPENDOCS,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,82,350,10
    CONTROL         "Replace: Don't move to the following occurrence",IDC_CHECK_REPLACEANDSTOP,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,97,350,10
    CONTROL         "Search Result window: show only one entry per found line",IDC_CHECK_SHOWONCEPERFOUNDLINE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,112,350,10
END
```

Reply:
* "When launch Find dialog" may not be ideal of meaning when the Find dialog is already open but the editor window has input focus (there's no "launch" because the window is already there).  More descriptive would be "When Find is initiated via menu or shortcut" but that is more words.
* "Fill find field with selected word" might be better as: "Fill find field with selected text" (because it could be more than a 'word' if user has more than that selected)
* "Select the word on which caret is" : I really think this needs to convey that this action only occurs where there is no pre-selected text, maybe "Select word under caret when nothing selected"

Sorry, Don, I'm not purposefully trying to change EVERY new line you've suggested for the UI.  :-)

I'm just trying to make the best UI possible. :-)

Perhaps solicit opinions on this from @pryrt @Yaron10 @vinsworldcom ... as a group we can make it the best!

Reply:
@donho - I like it!  I do also like @alankilborn suggestions of:

- "Fill find field with selected text"
- "Select word under caret when nothing selected"

Instead of "When launch Find dialog" or the more verbose but explicit, can we just use "Find Dialog Options"?

Cheers.

Reply:
Thank you for your suggestions.

Indeed, "When Find is initiated via menu or shortcut" is verbose.
How about "When Find dialog is invoked"? So it's implied both actions.

![image](https://user-images.githubusercontent.com/90293/186774229-1fa0e0cf-80af-4cc7-818c-9c25c31be578.png)

Note that if "Fill find field with selected text" is unchecked, then "Select word under caret when nothing selected" will also be unchecked (if checked) and disabled.

Do you guys see any scenario could be incoherent with the new GUI behaviour?

Reply:
I think the latest text is good enough; also the behavior seems consistent.

Reply:
I like the text changes.  My only thought is would it better to use "caret" or "cursor"?  I can make the changes when everyone is satisfied with the wording unless someone else is going to update the pull request.

Reply:
"caret" is the Scintilla term.

Reply:
@doug1234 
Let's use "cursor" instead of "caret".

Here are the codes you need copy-past:
```
#define IDD_PREFERENCE_SUB_SEARCHING 6900 //(IDD_PREFERENCE_BOX + 900)
	//#define IDC_CHECK_STOPFILLINGFINDFIELD (IDD_PREFERENCE_SUB_SEARCHING + 1)
	#define IDC_CHECK_MONOSPACEDFONT_FINDDLG (IDD_PREFERENCE_SUB_SEARCHING + 2)
	#define IDC_CHECK_FINDDLG_ALWAYS_VISIBLE (IDD_PREFERENCE_SUB_SEARCHING + 3)
	#define IDC_CHECK_CONFIRMREPLOPENDOCS (IDD_PREFERENCE_SUB_SEARCHING + 4)
	#define IDC_CHECK_REPLACEANDSTOP (IDD_PREFERENCE_SUB_SEARCHING + 5)
	#define IDC_CHECK_SHOWONCEPERFOUNDLINE (IDD_PREFERENCE_SUB_SEARCHING + 6)
	#define IDD_FILLFINDFIELD_GRPSTATIC (IDD_PREFERENCE_SUB_SEARCHING + 7)
	#define IDC_CHECK_FILLFINDFIEL_WITHSELECTED (IDD_PREFERENCE_SUB_SEARCHING + 8)
	#define IDC_CHECK_FILLFINDFIEL_SELECTCARET (IDD_PREFERENCE_SUB_SEARCHING + 9)
```
and
```
IDD_PREFERENCE_SUB_SEARCHING DIALOGEX 0, 0, 455, 185
STYLE DS_SETFONT | DS_FIXEDSYS | DS_CONTROL | WS_CHILD
FONT 8, "MS Shell Dlg", 0, 0, 0x1
BEGIN
    GROUPBOX        "When Find dialog is invoked",IDD_FILLFINDFIELD_GRPSTATIC,31,4,323,43,BS_CENTER
    CONTROL         "Fill find field with selected text",IDC_CHECK_FILLFINDFIEL_WITHSELECTED,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,16,275,10
    CONTROL         "Select word under cursor when nothing selected",IDC_CHECK_FILLFINDFIEL_SELECTCARET,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,52,31,275,10
    CONTROL         "Use Monospaced font in Find dialog (Need to restart Notepad++)",IDC_CHECK_MONOSPACEDFONT_FINDDLG,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,52,350,10
    CONTROL         "Find dialog remains open after search that outputs to results window",IDC_CHECK_FINDDLG_ALWAYS_VISIBLE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,67,350,10
    CONTROL         "Confirm Replace All in All Opened Documents",IDC_CHECK_CONFIRMREPLOPENDOCS,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,82,350,10
    CONTROL         "Replace: Don't move to the following occurrence",IDC_CHECK_REPLACEANDSTOP,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,97,350,10
    CONTROL         "Search Result window: show only one entry per found line",IDC_CHECK_SHOWONCEPERFOUNDLINE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,37,112,350,10
END
```
1. The default settings of this feature should be both "Fill find field with selected text" & "Select word under cursor when nothing selected" are checked.
2. The old settings should be remembered by this new feature: if "Don't fill find field in Find dialog with selected word" is checked in the old config, the new feature should keep both "Fill find field with selected text" & "Select word under cursor when nothing selected" unchecked. 
3. GUI behaviour: if "Fill find field with selected text" is unchecked, then "Select word under caret when nothing selected" will also be unchecked (if checked) and disabled.

Feel free to ask question if you have any.

Thank you for your coding effort.

Reply:
> Let's use "cursor" instead of "caret".

Oh, no, I'm absolutely HORRIFIED by this.
"Caret" is 100% the correct term (English is my only frame of reference, however).
Let's not introduce an inconsistency now.
Current usages in english.xml:
* `caret` : 3 hits
* `cursor` : 0 hits

Reply:
User manual:
* `caret` : 39 hits
* `cursor` : 22 hits
If there is agreement on "caret" for this PR, I'll put an issue in on the user manual to evaluate/change all instances of "cursor" to "caret".

Reply:
![image](https://user-images.githubusercontent.com/77065706/186794430-0fbb2989-b56a-494e-9240-6e8c9c051b06.png)

Reply:
![image](https://user-images.githubusercontent.com/77065706/186794527-0b5c27aa-cc3a-4c72-a437-331e3a725f00.png)

Reply:
I am OK with caret if it is what the majority wants.  I will try to get this wrapped up tomorrow night.  Thanks for the help!

Reply:
Since caret is used by scintilla, and everywhere in english.xml, that would be my vote, as well.

Reply:
It's true that "caret" is Scintilla term. However, the most users don't really use Scintilla directly.
And here is the definition of caret, which has nothing to do with the symbol that we are talking about: 
https://en.wikipedia.org/wiki/Caret_(computing)

Between "caret" and "cursor", I have no preference. All we should judge is from the aspect of the most of users.

Reply:
Notepad++ is my only editor of choice ;) 
But you guys may know which term the other editors use for this stuff?

Reply:
>which term the other editors use for this stuff?

VS2019 editor hasn't settled on one term (or...maybe it has and really is using the terms correctly, per Don's finding of "The cursor is the graphical image that represents the mouse..."):

![image](https://user-images.githubusercontent.com/77065706/186968806-ae92b652-c2e7-4eed-b07c-d4c9119ab3fc.png)

![image](https://user-images.githubusercontent.com/77065706/186969123-ccc64420-f822-471f-adf5-3c4e34ee820f.png)

Reply:
> VS2019 editor hasn't settled on one term (or...maybe it has and really is using the terms correctly

Properly speaking, a *caret* is an [insertion point][0], so more appropriate for what a text editor does, by metaphor with  the proof-reading symbol, an sub-scripted `^`.

A *cursor* points to or indexes a position (on the screen or in a file, like the "position indicator" in binary I/0 functions). The typical use case is (read-only) navigation, not editing.

Given their semantic and technical differences, there are perfectly valid reasons for using both terms in the same application.


[0]: https://www.merriam-webster.com/dictionary/caret

Reply:
>there are perfectly valid reasons for using both terms in the same application

OK, but there aren't currently any uses of "cursor" in the UI, and the usages of "caret" are all reasonable as that term.

The user manual is currently under revision to change any improper usages to the proper one, evaluating each individual "caret" and "cursor".

Reply:
OK. Let's keep using "caret".